### PR TITLE
Features

### PR DESCRIPTION
**Pull request by vanillainanutshell**
## What was done in this PR

* Add Book class
* Add Rentals class
* Established has-many relationships between Classroom and Student Classes
* Established many-many relationships between Persons and Book Classes using the intermediary Rental class.

### CODE REVIEW

File: classroom.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class ClassroomManager
+  acessor_attr :label, :students
```
Comment (GzimAsani) [inline classroom.rb]: - [ ] Kindly change this to should be `attr_accessor`

File: book.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+class Book
+  acessor_attr :title, :author
```
Comment (GzimAsani) [inline book.rb]: - [ ] Kindly change this to `attr_accessor`

File: book.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+  reader_attr :rentals
```
Comment (GzimAsani) [inline book.rb]: - [ ] Move the :rentals as attr_accessor as well

File: book.rb [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def create_rental(date, person)
+    Rental.new(date, safe, person)
```
Comment (GzimAsani) [inline book.rb]: - [ ] Please fix the **safe** typo here, it should be `self`

File: rental.rb [hunk 5]
```diff
@@ -0,0 +1,12 @@
+class Rental
+  attr_accessor :date
```
Comment (GzimAsani) [inline rental.rb]: - [ ] The `book` and `person` should also be in attr_accessor since they'll be used latter on your code

File: person.rb [hunk 6]
```diff
@@ -1,15 +1,17 @@
 require_relative 'nameable'
+require_relative 'rental'
 
 class Person < Nameable
-  reader_attr :id
+  reader_attr :id, :rentals
   accessor_attr :name, :age
```
Comment (GzimAsani) [inline person.rb]: - [ ] Please change this to `attr_accessor`
- [ ] Please add rentals in the `attr_accessor`

**Review by GzimAsani — CHANGES_REQUESTED** [review]
### Hi @vanillainanutshell 
 
 Good job so far! There are some issues that you still need to work on to go to the next project but you are almost there!
 
 ## Highlights
 * Your code organization is good.
 
 ## Required Changes 
 - [ ] _Check inline comments below :arrow_double_down: _




 _Check the comments under the review._
 
 ## Optional suggestions
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
 
Cheers and Happy coding! clapclapclap
 
 Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
 
 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: book.rb [hunk 7]
```diff
@@ -0,0 +1,13 @@
+class Book
+  attr_acessor :title, :author, :rentals
```
Comment (mmsesay) [inline book.rb]: - [ ] Please fix this typo. `attr_accessor` is the correct spelling. You only used one `c`

File: person.rb [hunk 8]
```diff
@@ -1,15 +1,17 @@
 require_relative 'nameable'
+require_relative 'rental'
 
 class Person < Nameable
-  reader_attr :id
-  accessor_attr :name, :age
+  attr_reader :id, :rentals
```
Comment (mmsesay) [inline person.rb]: - [ ] As suggested by the previous reviewer, please set the `:rentals` to attribute accessor

**Review by mmsesay — CHANGES_REQUESTED** [review]
Hi @vanillainanutshell, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

- [OPTIONAL] You should consider updating your readme description of your project.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by GzimAsani — APPROVED** [review]
# STATUS: APPROVED :100:         
## There is nothing else to say than it's time to merge... :partying_face: 
Cheers and Happy coding! :clap:  :computer:  :wine_glass: 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

### Please be reminded to not open a new pull request after you get a review, only use the same one :thought_balloon: 


 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**